benchmarks/bench_checks.py

```diff
@@ -15,9 +15,11 @@ def benchmark_check_grid_and_eko_compatible(test_files):
     wrong_grid = pineappl.grid.Grid.read(
         test_files / "data/grids/208/NUTEV_CC_NU_FE_SIGMARED.pineappl.lz4"
     )
-    ekoop = eko.output.legacy.load_tar(
-        test_files / "data/ekos/208/HERA_CC_318GEV_EM_SIGMARED.tar"
-    )
+    eko_filename = test_files / "data/ekos/208/HERA_CC_318GEV_EM_SIGMARED.tar"
+    if pineko.check.check_eko_is_legacy(eko_filename):
+        ekoop = eko.output.legacy.load_tar(eko_filename)
+    else:
+        ekoop = eko.output.EKO.load(eko_filename)
```

Comment:
Good, but just a suggestion: you are calling three functions with a single input (`eko_filename`), and this pattern is possibly going to repeat.

Just replace with a single call:
```suggestion
   ekoop = pineko.ekompatibility.load(eko_filename)
```

And define the function exactly in that way :)
(We could also move inside EKO, but I'd not go that far; since EKO is drastically moving in the new direction, let's keep compatibility in here as much as possible)